src/main/resources/assets/trickster/lavender/entries/tome_of_tomfoolery/basics/macros.md

```diff
@@ -0,0 +1,37 @@
+```json
+{
+  "title": "Macro Ring",
+  "icon": "trickster:macro_ring",
+  "category": "trickster:basics"
+}
+```
+
+Macros allow you to create your own Revisions to aid with spell scribing. 
+With a Macro Ring equipped, drawing a pattern that is a key in the ring will instead evaluate the spell associated with the key.
+
+;;;;;
+
+Macros take in the part of the spell that the pattern was drawn as the first argument. 
+The macro must then return a spell. This spell replaces the spell the pattern was drawn in.
+
+//TODO: add recipe
+
+;;;;;
+
+<|glyph@trickster:templates|trick-id=trickster:write_macro_ring,title=Macro Ploy|>
+
+{spell: spell} -> bool
```

Comment:
Oh wait, you already have an alternative! Yea this is way better.

Reply:
Yeah it was just `map` here and I thought that was silly, so I made them all into `{any: any}`, but didn't think to check anywhere except the `MapFragment`'s `Text` getter.